library/core/src/cmp.rs

```diff
@@ -1484,11 +1487,11 @@ mod impls {
     where
         A: ~const PartialEq<B>,
     {
-        #[inline]
+        #[inline(always)]
```

Comment:
Since inlining is bottom-up, `always` is usually not desirable on generic things, since it will inline-always the whole `eq` implementation if it happens to get inlined into this one.

These situations really want some kind of `inline(early)` that would inline the wrappers at MIR level without telling LLVM to [`alwaysinline`](https://llvm.org/docs/LangRef.html#function-attributes) them.